data-prepper-api/src/main/java/com/amazon/dataprepper/model/configuration/PipelineDescription.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.model.configuration;
+
+/**
+ * Model that should be provided to plugins to provide certain details about the pipeline they belong to
+ */
+public interface PipelineDescription {
```

Comment:
Please add `@since 1.3` to the Javadocs since this is a public interface.

I also think it is good to include it on the methods below.